server/src/test/java/org/apache/calcite/avatica/server/HttpServerSpnegoWithoutJaasTest.java

```diff
@@ -233,8 +233,8 @@ private static void setupUsers(File keytabDir) throws KrbException {
         // Passes the GSSCredential into the HTTP client implementation
         final AvaticaCommonsHttpClientImpl httpClient =
             new AvaticaCommonsHttpClientImpl(httpServerUrl);
-        httpClient.setGSSCredential(credential);
         httpClient.setHttpClientPool(pool, config);
+        httpClient.setGSSCredential(credential);
```

Comment:
Why swap these ?

Reply:
Inside setHttpClientPool, the initializeClient method is called, which initializes the persistentContext object. After that setGSSCredential method sets the credentials within persistentContext.
If we try to set the credentials before initializing persistentContext, it would result in a NullPointerException.